### PR TITLE
Conv: apply explicit padding after the transposition search (tf_converter)

### PR DESCRIPTION
**Pull request by tairenpiao**
### 1. Content and background

With `--tflite_backend tf_converter`, a Conv whose input has C == H == W and explicit padding can come out wrong while `-cotof` prints `Matches` for every OP.

When all non-batch axes of a Conv input have the same size, strict mode picks the input layout by trying every permutation that keeps the batch axis: transpose the validation data from the previous layer, run a dummy Conv on it, compare with the ONNX dummy inference, keep the smallest error. With explicit pads that can't be expressed as SAME, `input_tensor` is padded before that search but the validation data is not. For a 3x3 stride-2 Conv on 8x8 every candidate outputs (1,3,3,8) against ONNX's (1,8,4,4), all of them score the sentinel error, and `min_abs_err_perm_1` silently stays at identity. If the input is still in ONNX order, e.g. straight out of a Reshape, the Conv uses the wrong axis as channels. The per-OP check does not catch it because it tolerates layout ambiguity when the dims are equal.

### 2. Summary of corrections

`onnx2tf/ops/Conv.py` only; the padding decision is unchanged. I keep a reference to the unpadded tensor, apply the winning permutation to that, and pad afterwards. Each transposed validation candidate gets the same padding before the dummy Conv is built. Both go through `apply_explicit_padding`, which returns its argument unchanged unless `padded` is set, so every other path behaves exactly as before. Ties still resolve to identity (strict less-than, identity is the first candidate).

### 3. Before/After

Smallest model graph that shows it: `X(1,512) -> Add(const) -> Reshape(1,8,8,8) -> Conv(k3, s2, pads=[1,1,1,1]) -> Y(1,8,4,4)`. The Reshape leaves the Conv input in ONNX order; the Add makes the input non-constant, since the dummy inference feeds `np.ones` and a constant input gives every permutation the same output.

```python
import numpy as np, onnx
from onnx import TensorProto, helper, numpy_helper
rng = np.random.default_rng(0); N = 8
init = [numpy_helper.from_array(rng.standard_normal((N, N, 3, 3)).astype(np.float32), "W"),
        numpy_helper.from_array(rng.standard_normal((N,)).astype(np.float32), "B"),
        numpy_helper.from_array(rng.standard_normal((N*N*N,)).astype(np.float32), "bias512"),
        numpy_helper.from_array(np.array([1, N, N, N], dtype=np.int64), "shape")]
g = helper.make_graph(
    [helper.make_node("Add", ["X", "bias512"], ["A"]),
     helper.make_node("Reshape", ["A", "shape"], ["R"]),
     helper.make_node("Conv", ["R", "W", "B"], ["Y"], kernel_shape=[3,3], strides=[2,2], pads=[1,1,1,1])],
    "conv_padded_same_axes",
    [helper.make_tensor_value_info("X", TensorProto.FLOAT, [1, N*N*N])],
    [helper.make_tensor_value_info("Y", TensorProto.FLOAT, [1, N, N//2, N//2])], init)
m = helper.make_model(g, opset_imports=[helper.make_opsetid("", 13)]); m.ir_version = 8
onnx.save(m, "conv_padded_same_axes.onnx")
```

```
onnx2tf -i conv_padded_same_axes.onnx -o out -tb tf_converter -nuo -cotof
```

main (d6cd7159), with a temporary print in the search loop:

```
perm=(0, 1, 2, 3) err=9223372036854775807 tf_out=(1, 3, 3, 8) onnx=(1, 8, 4, 4)
perm=(0, 1, 3, 2) err=9223372036854775807 tf_out=(1, 3, 3, 8) onnx=(1, 8, 4, 4)
perm=(0, 2, 1, 3) err=9223372036854775807 tf_out=(1, 3, 3, 8) onnx=(1, 8, 4, 4)
perm=(0, 2, 3, 1) err=9223372036854775807 tf_out=(1, 3, 3, 8) onnx=(1, 8, 4, 4)
perm=(0, 3, 1, 2) err=9223372036854775807 tf_out=(1, 3, 3, 8) onnx=(1, 8, 4, 4)
perm=(0, 3, 2, 1) err=9223372036854775807 tf_out=(1, 3, 3, 8) onnx=(1, 8, 4, 4)
chosen perm=[0, 1, 2, 3]
INFO: onnx_output_name: Y shape: (1, 8, 4, 4) dtype: float32 validate_result:  Matches
ONNX/TFLite output check complete! (out/conv_padded_same_axes_accuracy_report.json) trigger=-cotof(auto) source=tf_converter target=float32 max_abs=57.625 rmse=16.7022 cosine=0.154714 pass=False
```

this PR:

```
perm=(0, 1, 2, 3) err=41.81199645996094     tf_out=(1, 4, 4, 8)
perm=(0, 1, 3, 2) err=30.937366485595703    tf_out=(1, 4, 4, 8)
perm=(0, 2, 1, 3) err=27.101259231567383    tf_out=(1, 4, 4, 8)
perm=(0, 2, 3, 1) err=3.814697265625e-06    tf_out=(1, 4, 4, 8)
chosen perm=[0, 2, 3, 1]
ONNX/TFLite output check complete! (out/conv_padded_same_axes_accuracy_report.json) trigger=-cotof(auto) source=tf_converter target=float32 max_abs=0 rmse=0 cosine=1 pass=True
```

Exported float32 TFLite against the ONNX model on the same random input, TF output transposed to NCHW:

| | permutation search | `-cotof` output check | float32 TFLite vs ONNX |
|---|---|---|---|
| main (d6cd7159) | all 6 candidates hit the sentinel error, identity picked | `max_abs=57.6 cosine=0.15 pass=False` | SNR -2.98 dB, max abs error 46.0 |
| this PR | `[0,2,3,1]` err 3.8e-06, identity err 41.8 | `max_abs=0 cosine=1 pass=True` | SNR 342 dB, bit-identical |

Regression check: a (1,8,8,8) input fed straight into the Conv with pads [1,1,1,1] and with [1,1,0,0], and the Reshape model without the Add (constant input), all give a float32 TFLite that is bit-identical to what main produces, and the per-OP check still says `Matches`. The constant-input Reshape model is wrong on both, since `np.ones` dummy data cannot separate the permutations; that is unrelated to this change. The same model with `-dsm` still pads eagerly and outputs (1,4,4,8), and the default `flatbuffer_direct` backend is untouched. Python 3.12.11, tensorflow 2.21.0, onnx 1.20.1, onnxruntime 1.26.0, macOS arm64.

### 4. Issue number

None.

### CODE REVIEW

**Review by PINTO0309 — COMMENTED** [review]
Reviewed c8dce14. The fix looks correct: both the transposition search and the final graph now apply transpose → explicit padding → Conv, so candidates are evaluated under the same conditions as the generated graph. I found no blocking issues in the cases tested.

I compared Keras outputs against ONNX Runtime across 16 cases, including Conv1D/3D, depthwise and grouped convolutions, asymmetric padding, zero/SAME padding, and disabled strict mode. Nine cases improved; the remaining seven produced exactly the same outputs before and after the change. The existing all-ones validation limitation remains unchanged.

For the reported reproduction model, I also exported float32 TFLite and tested three independent random inputs. The maximum absolute error against ONNX Runtime decreased from 51.20 to 3.81e-6 on Linux with TensorFlow 2.21.0 and ONNX Runtime 1.26.0.

Two non-blocking suggestions:

- Please add an automated regression test using `Add(nonuniform constant) → Reshape → Conv`, comparing the final output against ONNX Runtime on independent random inputs. This would catch the original failure without relying on per-operator validation messages. Symmetric and asymmetric explicit padding would be useful cases.
- Please keep the description's “all operators report Matches” and “bit-identical” observations scoped to the reported environment. On Linux, the original Conv output was reported as `Unmatched`, and the fixed TFLite output had the small error noted above. These differences do not affect the validity of the fix.

**Comment by PINTO0309** [comment]
Thanks
